module-1/lab-functional-programming/your-code/Q1.ipynb

```diff
@@ -19,7 +19,7 @@
   },
```

Comment:
Really good approach Alberto!! Well done using list comprehensions and making your code shorter. 
One recommendation on list comprehensions with two levels, the work fine always but make the code less readable. I would suggest in this cases write down a for loop with two levels. This could be a way, change the variables names with yours in your code. 
```Python
    for s in corpus:
        s = ''.join(c for c in s if c not in punctuation)
        terms = s.split()
        for term in terms:
            if not term in bag_of_words and not term in stop_words:
                bag_of_words.append(term)
   ```